### ISSUE TITLE
request method fail with bulkImport

### ISSUE THREAD

**Issue by clementtrebuchet** [HSRNetwork/piapi#3]
**_Calling_**
api.request('bulkImport', params=json.dumps(data))

*_When making a PUT request on bulkImport *_
Traceback (most recent call last):
Desktop/Code/inno1_configuration_manager/src/configuration_manager/core/provisionners/pr_con.py", line 64, in <module>
    PrimeCon(data=[{}], config=d)._update_api()
Desktop/Code/inno1_configuration_manager/src/configuration_manager/core/provisionners/pr_con.py", line 49, in _update_api
2016-02-25 16:39:18,602 - requests.packages.urllib3.connectionpool - DEBUG - "PUT /webacs/api/v1/op/devices/bulkImport.json HTTP/1.1" 500 373
    api.request('bulkImport', params=json.dumps(data))
  File "C:\Python27\lib\site-packages\piapi.py", line 394, in request
    return self.request_service(resource, params, timeout)
  File "C:\Python27\lib\site-packages\piapi.py", line 354, in request_service
    return self._parse(response)
  File "C:\Python27\lib\site-packages\piapi.py", line 178, in _parse
    raise PIAPIRequestError("An error has occured during the API invocation")
piapi.PIAPIRequestError: An error has occured during the API invocation

**if i modify request_service method with**
elif method == "PUT":
            headers  = {"Content-Type": "application/json; charset=UTF-8"}
            response = self.session.request(method, url, data=params, headers=headers, verify=self.verify,
                                            timeout=timeout)
**_then the request pass_**

2016-02-25 16:43:59,295 - requests.packages.urllib3.connectionpool - DEBUG - "PUT /webacs/api/v1/op/devices/bulkImport.json HTTP/1.1" 200 313

**Comment by maximumG** [comment]
Thanks for pointing this out @clementtrebuchet . I will modify the behaviour in **all** requests made to Cisco Prime API even the GET ones in order not to have to call 'resource'**.json** each time in the URL.

**Comment by baileytj3** [comment]
I just ran across this issue myself. I made the change that @clementtrebuchet suggested but I was still receiving a 500 response code from Prime. I had to make the following change for Prime to accept the API call.

```
elif method == "PUT":
  headers = {"Content-Type": "application/json; charset=UTF-8"}
  response = self.session.request(method, url, data=json.dumps(params), headers=headers, verify=self.verify, timeout=timeout)
```

Im assuming it has something to do with the way that Python's str() method works on dictionaries.

```
>>> data
{'accessPoint': {'adminStatus': 'ENABLE', 'unifiedApInfo': {'primaryMwar': 'controller1', 'secondaryMwar': 'controller2', 'tertiaryMwar': 'controller3'}, 'mapLocation': 'default location', 'name': 'TestNaming', 'accessPointId': 1234}}
```

vs

```
>>> json.dumps(data)
'{"accessPoint": {"adminStatus": "ENABLE", "unifiedApInfo": {"primaryMwar": "controller1", "secondaryMwar": "controller2", "tertiaryMwar": "controller3"}, "mapLocation": "default location", "name": "TestNaming", "accessPointId": 1234}}'
```

**Comment by maximumG** [comment]
Great to see that some folks are contributing to the library 👍 .

@baileytj3 : would it be possible to create a merge request with your changes ? thanks

**Comment by msom** [comment]
Probably fixed with 5189b156a04bcab7a9d1a56a08184e721401f888
